core/rfb.js

```diff
@@ -398,11 +398,22 @@ export default class RFB extends EventTargetMixin {
     _connect() {
         Log.Debug(">> RFB.connect");
 
-        Log.Info("connecting to " + this._url);
-
         try {
-            // WebSocket.onopen transitions to the RFB init states
-            this._sock.open(this._url, this._wsProtocols);
+            // WebChannel.onopen transitions to the RFB init states
+            if (typeof this._urlOrChannel === "string") {
```

Comment:
You might want to split `RFB._connect()` so that the "url or object" check can be done already in the constructor.